diff --git a/src/valkey/valkey.zig b/src/valkey/valkey.zig
index c161449..7af44a0 100644
--- a/src/valkey/valkey.zig
+++ b/src/valkey/valkey.zig
@@ -934,7 +934,7 @@ pub const ValkeyClient = struct {
     }
 
     inline fn parent(this: *ValkeyClient) *JSValkeyClient {
-        return @fieldParentPtr("client", this);
+        return @alignCast(@fieldParentPtr("client", this));
     }
 
     inline fn globalObject(this: *ValkeyClient) *JSC.JSGlobalObject {
