diff --git a/deflate.c b/deflate.c
index d861faf..e025094 100644
--- a/deflate.c
+++ b/deflate.c
@@ -145,10 +145,11 @@ static uint32_t hash_func(deflate_state *s, void* str) {
 }
 
 #else
-
-#error "Only 64-bit Intel and ARM architectures are supported"
-
-#endif
+#if defined(__arm__) || defined(__thumb__) || defined(_M_ARM)
+static uint32_t hash_func(deflate_state *s, void* str) {
+    return crc32(0, (const unsigned char *)str, 4) & s->hash_mask;
+}
+#else
 
 /* ===========================================================================
  * Insert string str in the dictionary and return the previous head
  * of the hash chain (the most recent string with same hash key).
```