diff --git a/src/bun.js/bindings/JSBuffer.cpp b/src/bun.js/bindings/JSBuffer.cpp
index 2e6479a..211968a 100644
--- a/src/bun.js/bindings/JSBuffer.cpp
+++ b/src/bun.js/bindings/JSBuffer.cpp
@@ -1575,8 +1575,8 @@ static JSC::EncodedJSValue jsBufferPrototypeFunction_inspectBody(JSC::JSGlobalOb
     for (auto item : data.first(actualMax)) {
         any = true;
         result.append(' ');
-        result.append(alphabet[item / 16]);
-        result.append(alphabet[item % 16]);
+        result.append(alphabet[static_cast<size_t>(item / 16)]);
+        result.append(alphabet[static_cast<size_t>(item % 16)]);
     }
     if (data.size() > max) {
         auto remaining = data.size() - max;
@@ -2229,7 +2229,11 @@ JSC_DEFINE_JIT_OPERATION(jsBufferConstructorAllocWithoutTypeChecks, JSUint8Array
     CallFrame* callFrame = DECLARE_CALL_FRAME(vm);
     IGNORE_WARNINGS_END
     JSC::JITOperationPrologueCallFrameTracer tracer(vm, callFrame);
+#if USE(JSVALUE64)
     return { allocBuffer(lexicalGlobalObject, byteLength) };
+#else
+    return asExceptionResult(allocBuffer(lexicalGlobalObject, byteLength), nullptr);
+#endif
 }
 
 JSC_DEFINE_JIT_OPERATION(jsBufferConstructorAllocUnsafeWithoutTypeChecks, JSUint8Array*, (JSC::JSGlobalObject * lexicalGlobalObject, void* thisValue, int byteLength))
@@ -2239,7 +2243,11 @@ JSC_DEFINE_JIT_OPERATION(jsBufferConstructorAllocUnsafeWithoutTypeChecks, JSUint
     CallFrame* callFrame = DECLARE_CALL_FRAME(vm);
     IGNORE_WARNINGS_END
     JSC::JITOperationPrologueCallFrameTracer tracer(vm, callFrame);
+#if USE(JSVALUE64)
     return { allocBufferUnsafe(lexicalGlobalObject, byteLength) };
+#else
+    return asExceptionResult(allocBufferUnsafe(lexicalGlobalObject, byteLength), nullptr);
+#endif
 }
 
 JSC_DEFINE_JIT_OPERATION(jsBufferConstructorAllocUnsafeSlowWithoutTypeChecks, JSUint8Array*, (JSC::JSGlobalObject * lexicalGlobalObject, void* thisValue, int byteLength))
@@ -2249,7 +2257,11 @@ JSC_DEFINE_JIT_OPERATION(jsBufferConstructorAllocUnsafeSlowWithoutTypeChecks, JS
     CallFrame* callFrame = DECLARE_CALL_FRAME(vm);
     IGNORE_WARNINGS_END
     JSC::JITOperationPrologueCallFrameTracer tracer(vm, callFrame);
+#if USE(JSVALUE64)
     return { allocBufferUnsafe(lexicalGlobalObject, byteLength) };
+#else
+    return asExceptionResult(allocBufferUnsafe(lexicalGlobalObject, byteLength), nullptr);
+#endif
 }
 
 JSC_ANNOTATE_HOST_FUNCTION(JSBufferConstructorConstruct, JSBufferConstructor::construct);
